app/scripts/directives/nav.js

```diff
@@ -204,6 +204,19 @@ angular.module('openshiftConsole')
     return {
       restrict: 'EA',
       templateUrl: 'views/directives/header/header.html',
+      controller: function (AuthService, HomePagePreference) {
+        AuthService.withUser().then(function() {
+          HomePagePreference.init($rootScope.user);
+          switch(HomePagePreference.getHomePagePreference()) {
+            case 'project-list':
+              $location.path("/projects");
```

Comment:
We can't unconditionally redirect. This will break direct linking to any URL in the web console! We should only use the value of path is empty.

I'm don't think the nav controller is the right place to make this check. Is it possible to do the check here?

https://github.com/openshift/origin-web-console/blob/master/app/scripts/app.js#L39-L59

Then add another `/catalog` route for the catalog that is used when you click the logo?

Reply:
```js
Naviage.toProjectList();
```

Reply:
How would you recommend navigating to '/project/my-project/overview' in app.js's .config block when the preferred home page is set to a user's project overview?  -thanks

Reply:
This seems to work:
```
.run(function(HomePagePreference, Navigate) {
    var homePagePref = HomePagePreference.getHomePagePreference();
    if(homePagePref === "project-overview"){
      Navigate.toProjectOverview(HomePagePreference.getProjectOverviewPage());
    }
  })
```

Reply:
> How would you recommend navigating to '/project/my-project/overview' in app.js's .config block when the preferred home page is set to a user's project overview?

Check the home page preference once in that block (if the user is logged in) and then 

```js
var homePagePath = <determine home page path...>
if (homePagePath) {
  $routeProvider.when('/', { redirectTo: homePagePath } );
} else {
  $routeProvider.when('/', {
    templateUrl: 'views/landing-page.html',
    controller: 'LandingPageController',
    reloadOnSearch: false
  });
}
```

Reply:
> I don't think the nav controller is the right place to make this check. Is it possible to do the check here?
> 
> https://github.com/openshift/origin-web-console/blob/master/app/scripts/app.js#L39-L59

Hi @spadgett,  the problem I'm running into is that the user is not logged in at the time app.js is executed.   I believed you and @jwforres indicated if getting the username was going to be an issue, then just store the homePagePref per browser and not use the userName in the key.   File a follow up issue to track.   Are we at that point? :-)

Reply:
I'm OK with that.

One thing you could try, though, is to register an `AuthService.onUserChanged` callback in app.js and update the route to redirect if the user has a home page preference.

Reply:
Ok, so I have `AuthService.onUserChanged` working in app.js's .run block.   The only issue is that when user sets a new home page, clicking on the logo will still goto the old route/prev. home page until the user logs in again or reloads "/" in the browser (so that app.js .run/.config blocks are rerun to reset the `$routeParams`.